Describe defaults of more options

jablko opened the pull request:
Adds `defaultValueDescription` to:
- importsNotUsedAsValues
- watchFile
- watchDirectory
- fallbackPolling

/cc @orta

sandersn commented:
This PR seems to do a lot more than add default value descriptions. Can you add more explanation to cover the rest of the changes?

jablko commented:
:+1: Can do:
1. In 9c2740badf536c48ba031e53e6423f32d1b43cc0 I added `defaultValueDescription`s to `importsNotUsedAsValues`, `watchFile`, `watchDirectory` and `fallbackPolling`.
2. I revised the `CommandLineOptionBase` and `CommandLineOptionOfCustomType` types, in https://github.com/microsoft/TypeScript/pull/46498/commits/61b07a38349198a6b234ee7ddb780727c257b709#diff-e9fd483341eea176a38fbd370590e1dc65ce2d9bf70bfd317c5407f04dba9560R6376, to accommodate enum members (e.g. `defaultValueDescription: ImportsNotUsedAsValues.Remove`) and analogously revised `CommandLineOptionOfNumberType` and `CommandLineOptionOfBooleanType` from strings -> numbers/booleans (e.g. `` `${number | undefined}` `` -> `number | undefined`.
3. I updated --help (https://github.com/microsoft/TypeScript/pull/46498/commits/61b07a38349198a6b234ee7ddb780727c257b709#diff-5bc64df2f0c1746eae137760ddd218696b382812e3b408d417bd838f9620a75dR212) to stringify enum members (`ImportsNotUsedAsValues.Remove` -> `remove` vs. e.g. `0`).

orta reviewed (approved):
Thanks for the writeup, this looks good to me and I think it improves the CLI UX 👍🏻
